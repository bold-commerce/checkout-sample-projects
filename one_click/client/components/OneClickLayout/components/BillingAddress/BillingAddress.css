.FieldSet--BillingAddress .RadioButton .stx-field {
  position: unset;
}
.FieldSet--BillingAddress .RadioButton .stx-radio {
  position: unset;
}

.FieldSet--BillingAddress.FieldSet {
  display: flex;
  flex-direction: column;
  margin: 0;
}

.FieldSet--BillingAddress .RadioButton {
  margin-bottom: 0;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  display: flex;
  align-items: center;
  padding-left: 0;
  cursor: pointer;
}

.FieldSet--BillingAddress .FieldSet__Content {
  display: grid;
  grid-auto-flow: row;
  grid-gap: 10px;
}

.FieldSet--BillingAddress .RadioButton .stx-checkbox-field__checkbox {
  position: relative;
  left: 0;
  top: 0;
  margin-right: 10px;
}

.FieldSet--BillingAddress .RadioButton .stx-radio-field__label {
  position: absolute;
  left: 0;
  height: 100%;
  top: 0;
  padding-left: 45px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  cursor: pointer;
  width: 100%;
}
