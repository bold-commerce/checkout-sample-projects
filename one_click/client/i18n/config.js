import i18n from 'i18next';
import enTranslation from './en/translation.json';
import { initReactI18next } from 'react-i18next';

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: {
        translation: enTranslation
      }
    },
    lng: 'en',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false, 
    }
  });

export default i18n;