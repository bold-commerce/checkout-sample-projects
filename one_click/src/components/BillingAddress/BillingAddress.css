.FieldSet--BillingAddress .RadioButton .stx-field {
  position: unset;
}
.FieldSet--BillingAddress .RadioButton .stx-radio {
  position: unset;
}

.FieldSet--BillingAddress.FieldSet {
  display: flex;
  flex-direction: column;
  margin: 0;
}

.FieldSet--BillingAddress .RadioButton {
  margin-bottom: 0;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  display: flex;
  align-items: center;
  padding-left: 0;
  cursor: pointer;
}

.FieldSet--BillingAddress .FieldSet__Content .stx-checkbox-field__div {
  display: flex;
}

.FieldSet--BillingAddress .RadioButton .stx-checkbox-field__checkbox {
  position: relative;
  margin-right: 10px;
}

.FieldSet--BillingAddress .RadioButton .stx-checkbox-field__label {
  height: 100%;
  cursor: pointer;
  width: 100%;
  color: #000;
}

.Billing__Title {
  font-size: 14px;
  font-weight: bold;
  line-height: 20px;
  color: rgba(0, 0, 0, .9);
  margin-bottom: 4px;
}