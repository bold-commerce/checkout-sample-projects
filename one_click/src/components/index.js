export * from './Address';
export * from './BackButton';
export * from './Billing';
export * from './BillingAddress';
export * from './Card';
export * from './CheckoutButton';
export * from './CheckoutForm';
export * from './Confirmation';
export * from './CustomerInfo';
export * from './Discount';
export * from './EmptyState';
export * from './Header';
export * from './Icons';
export * from './Index';
export * from './InventoryItem';
export * from './LineItems';
export * from './LoadingState';
export * from './Payment';
export * from '../pages/Processing';
export * from './Product';
export * from './RadioItem';
export * from './Shipping';
export * from './ShippingLines';
export * from './Summary';
