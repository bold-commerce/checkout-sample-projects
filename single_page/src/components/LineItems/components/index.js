export { default as LineItem } from './LineItem';
export { default as Product } from './Product';
