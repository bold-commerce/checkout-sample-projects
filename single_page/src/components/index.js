export * from './Address';
export * from './BillingAddress';
export * from './CheckoutButton';
export * from './CheckoutSection';
export * from './Customer';
export * from './DiscountForm';
export * from './EmptyState';
export * from './Header';
export * from './LineItems';
export * from './LoadingState';
export * from './OrderErrors';
export * from './OrderSummary';
export * from './PaymentMethod';
export * from './ShippingAddress';
export * from './ShippingLines';
