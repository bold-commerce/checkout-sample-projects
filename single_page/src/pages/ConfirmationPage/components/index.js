export * from './ConfirmationList';
export * from './ConfirmationListItem';
