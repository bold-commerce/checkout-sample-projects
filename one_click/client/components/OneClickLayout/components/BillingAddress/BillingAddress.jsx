/* eslint-disable react/forbid-prop-types */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { CheckboxField } from '@boldcommerce/stacks-ui';
import { Address } from '../Address';
import { useBillingAddress, useBillingSameAsShipping, useCountryInfo } from '@boldcommerce/checkout-react-components';
import './BillingAddress.css';

export const BillingAddress = ({
  billingAddress, billingSameAsShipping, submitBillingAddress, setBillingSameAsShipping, onChange, requiredAddressFields,
}) => {
  const [address, setAddress] = useState(billingAddress);
  const { data } = useCountryInfo(address);
  const{
    countries,
    provinces,
    showProvince,
    showPostalCode,
    provinceLabel,
  } = data;
  const [loading, setLoading] = useState(false);
  const [errors, setErrors] = useState(null);

  let provincePlaceholder = provinceLabel;

  // TODO: replace with languages config file
  if (provinceLabel === 'state_territory') {
    provincePlaceholder = 'state/territory';
  }

  const handleSubmit = async(data) => {
    setLoading(true);
    try {
      await submitBillingAddress(data);
      setErrors(null);
    } catch(e) {
      setErrors(e.body.errors);
    }
    setLoading(false);
  };

  const handleSetBillingSameAsShipping = async(data) => {
    setLoading(true);
    try {
      await setBillingSameAsShipping(data)
      setErrors(null);
    } catch(e) {
      setErrors(e.body.errors);
    }
    setLoading(false);
  };

  return (
    <section className="FieldSet FieldSet--BillingAddress">
      <div className="FieldSet__Header">
        <h3>Billing address</h3>
      </div>
      <div className="FieldSet__Content">
        <CheckboxField
          label="Same as shipping address"
          name="billing-address"
          className="RadioButton"
          value="SAME_AS_SHIPPING_ADDRESS"
          checked={billingSameAsShipping}
          disabled={loading}
          onChange={() => handleSetBillingSameAsShipping(!billingSameAsShipping)}
        />
        { billingSameAsShipping ? null
          : (
            <Address
              address={address}
              onChange={(data) => setAddress((prevAddress) => ({
                ...prevAddress,
                ...data,
              }))}
              errors={errors}
              countries={countries}
              provinces={provinces}
              showPostalCode={showPostalCode}
              showProvince={showProvince}
              provinceLabel={provincePlaceholder}
              submit={() => handleSubmit(address)}
              requiredAddressFields={requiredAddressFields}
            />
          )}
      </div>
    </section>
  );
};

BillingAddress.propTypes = {
  billingAddress: PropTypes.any,
  countryInfo: PropTypes.array,
  billingAddressErrors: PropTypes.object,
  billingSameAsShipping: PropTypes.bool,
  submitBillingAddress: PropTypes.func,
  setBillingSameAsShipping: PropTypes.func,
  onChange: PropTypes.func,
  requiredAddressFields: PropTypes.array,
};

const MemoizedBillingAddress = React.memo(BillingAddress);

const BillingAddressContainer = ({ requiredAddressFields }) => {
  const { data: billingAddress, submitBillingAddress } = useBillingAddress();
  const { data: billingSameAsShipping, setBillingSameAsShipping } = useBillingSameAsShipping(); 

  return (
    <MemoizedBillingAddress
      billingAddress={billingAddress}
      billingSameAsShipping={billingSameAsShipping}
      submitBillingAddress={submitBillingAddress}
      setBillingSameAsShipping={setBillingSameAsShipping}
      requiredAddressFields={requiredAddressFields}
    />
  );
};

BillingAddressContainer.propTypes = {
  requiredAddressFields: PropTypes.array,
};

export default BillingAddressContainer;
