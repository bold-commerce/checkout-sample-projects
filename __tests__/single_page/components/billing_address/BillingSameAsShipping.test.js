import React from 'react';
import { render } from "@testing-library/react";
import { BillingSameAsShipping } from "../../../../single_page/src/components/BillingAddress";
import '../../../../single_page/src/i18n/config'

describe('BillingSameAsShipping', () => {
  test('renders BillingSameAsShipping component with billing the same as shipping', () => {
    const { asFragment } = render( 
      <BillingSameAsShipping billingSameAsShipping />
    );

    expect(asFragment()).toMatchSnapshot();
  })
  test('renders BillingSameAsShipping component with billing not the same as shipping', () => {
    const { asFragment } = render( 
      <BillingSameAsShipping />
    );

    expect(asFragment()).toMatchSnapshot();
  })
  test('renders BillingSameAsShipping component disabled with billing same as shipping', () => {
    const { asFragment } = render( 
      <BillingSameAsShipping billingSameAsShipping disabled />
    );

    expect(asFragment()).toMatchSnapshot();
  })
  test('renders BillingSameAsShipping component disabled with billing not the same as shipping', () => {
    const { asFragment } = render( 
      <BillingSameAsShipping disabled />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
