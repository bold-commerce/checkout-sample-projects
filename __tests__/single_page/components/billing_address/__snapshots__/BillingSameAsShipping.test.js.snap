// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillingSameAsShipping renders BillingSameAsShipping component disabled with billing not the same as shipping 1`] = `
<DocumentFragment>
  <div
    class="stx-field stx-field--is-disabled stx-field--with-radio-field RadioButton"
  >
    <div
      class="stx-radio-field"
    >
      <span
        class="stx-radio stx-radio--is-disabled stx-radio-field__radio"
      >
        <input
          aria-disabled="true"
          class="stx-radio__input"
          disabled=""
          id="stxField7"
          name="billing-address"
          role="radio"
          type="radio"
          value="SAME_AS_SHIPPING_ADDRESS"
        />
        <span
          class="stx-radio__box"
        />
      </span>
      <label
        class="stx-radio-field__label"
        for="stxField7"
      >
        Same as shipping address
      </label>
    </div>
  </div>
  <div
    class="stx-field stx-field--is-disabled stx-field--with-radio-field RadioButton"
  >
    <div
      class="stx-radio-field"
    >
      <span
        class="stx-radio stx-radio--is-disabled stx-radio--is-checked stx-radio-field__radio"
      >
        <input
          aria-checked="true"
          aria-disabled="true"
          checked=""
          class="stx-radio__input"
          disabled=""
          id="stxField8"
          name="billing-address"
          role="radio"
          type="radio"
          value="DIFFERENT_BILLING_ADDRESS"
        />
        <span
          class="stx-radio__box"
        />
      </span>
      <label
        class="stx-radio-field__label"
        for="stxField8"
      >
        Use a different billing address
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BillingSameAsShipping renders BillingSameAsShipping component disabled with billing same as shipping 1`] = `
<DocumentFragment>
  <div
    class="stx-field stx-field--is-disabled stx-field--with-radio-field RadioButton"
  >
    <div
      class="stx-radio-field"
    >
      <span
        class="stx-radio stx-radio--is-disabled stx-radio--is-checked stx-radio-field__radio"
      >
        <input
          aria-checked="true"
          aria-disabled="true"
          checked=""
          class="stx-radio__input"
          disabled=""
          id="stxField5"
          name="billing-address"
          role="radio"
          type="radio"
          value="SAME_AS_SHIPPING_ADDRESS"
        />
        <span
          class="stx-radio__box"
        />
      </span>
      <label
        class="stx-radio-field__label"
        for="stxField5"
      >
        Same as shipping address
      </label>
    </div>
  </div>
  <div
    class="stx-field stx-field--is-disabled stx-field--with-radio-field RadioButton"
  >
    <div
      class="stx-radio-field"
    >
      <span
        class="stx-radio stx-radio--is-disabled stx-radio-field__radio"
      >
        <input
          aria-checked="false"
          aria-disabled="true"
          class="stx-radio__input"
          disabled=""
          id="stxField6"
          name="billing-address"
          role="radio"
          type="radio"
          value="DIFFERENT_BILLING_ADDRESS"
        />
        <span
          class="stx-radio__box"
        />
      </span>
      <label
        class="stx-radio-field__label"
        for="stxField6"
      >
        Use a different billing address
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BillingSameAsShipping renders BillingSameAsShipping component with billing not the same as shipping 1`] = `
<DocumentFragment>
  <div
    class="stx-field stx-field--with-radio-field RadioButton"
  >
    <div
      class="stx-radio-field"
    >
      <span
        class="stx-radio stx-radio-field__radio"
      >
        <input
          class="stx-radio__input"
          id="stxField3"
          name="billing-address"
          role="radio"
          type="radio"
          value="SAME_AS_SHIPPING_ADDRESS"
        />
        <span
          class="stx-radio__box"
        />
      </span>
      <label
        class="stx-radio-field__label"
        for="stxField3"
      >
        Same as shipping address
      </label>
    </div>
  </div>
  <div
    class="stx-field stx-field--with-radio-field RadioButton"
  >
    <div
      class="stx-radio-field"
    >
      <span
        class="stx-radio stx-radio--is-checked stx-radio-field__radio"
      >
        <input
          aria-checked="true"
          checked=""
          class="stx-radio__input"
          id="stxField4"
          name="billing-address"
          role="radio"
          type="radio"
          value="DIFFERENT_BILLING_ADDRESS"
        />
        <span
          class="stx-radio__box"
        />
      </span>
      <label
        class="stx-radio-field__label"
        for="stxField4"
      >
        Use a different billing address
      </label>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BillingSameAsShipping renders BillingSameAsShipping component with billing the same as shipping 1`] = `
<DocumentFragment>
  <div
    class="stx-field stx-field--with-radio-field RadioButton"
  >
    <div
      class="stx-radio-field"
    >
      <span
        class="stx-radio stx-radio--is-checked stx-radio-field__radio"
      >
        <input
          aria-checked="true"
          checked=""
          class="stx-radio__input"
          id="stxField1"
          name="billing-address"
          role="radio"
          type="radio"
          value="SAME_AS_SHIPPING_ADDRESS"
        />
        <span
          class="stx-radio__box"
        />
      </span>
      <label
        class="stx-radio-field__label"
        for="stxField1"
      >
        Same as shipping address
      </label>
    </div>
  </div>
  <div
    class="stx-field stx-field--with-radio-field RadioButton"
  >
    <div
      class="stx-radio-field"
    >
      <span
        class="stx-radio stx-radio-field__radio"
      >
        <input
          aria-checked="false"
          class="stx-radio__input"
          id="stxField2"
          name="billing-address"
          role="radio"
          type="radio"
          value="DIFFERENT_BILLING_ADDRESS"
        />
        <span
          class="stx-radio__box"
        />
      </span>
      <label
        class="stx-radio-field__label"
        for="stxField2"
      >
        Use a different billing address
      </label>
    </div>
  </div>
</DocumentFragment>
`;
